### DISCUSSION TITLE
Prediction - unspecialized (& incremental)

### DISCUSSION THREAD

**Discussion by doubledh** [Imaginary-Horizons-Productions/prophets-of-the-labyrinth-Archive#330] [Ideas]
Prediction is an abstracted mechanic, implemented differently, uniquely, and mutually exclusively between archetypes. Sometimes it can be difficult to get by with the information allowed to the current party formation (especially for parties of fewer members)

Therefore, I propose that rather than restricting a certain prediction to a single archetype, that all archetypes can predict all information, but each archetype is either more or less efficient for certain metrics, than others (specialized)

With the ability to do unspecialized prediction, it may ALSO be possible to predict information over the course of battle, and have predicted values ACCUMULATE, rather than be evaluated at each turn. In this way, a battle could have a finite amount of information for each of the combatants, and that information is unlocked per turn.

With incremental predictions, it would work well in conjunction with a [unified prediction display](https://github.com/Imaginary-Horizons-Productions/prophets-of-the-labyrinth/discussions/331)

**Comment by doubledh** [comment]
one comfier balanced proposal:
each class has one predict that they are specialized in, and two that they are unspecialized in, and cannot predict others beyond that.
specialized either means that they can predict one additional instance for free per turn in addition to another predict, or that if they predict in their specialty, they do it in a blanket fashion.

I like the one additional instance for free per turn for specialized predict

That would mean that the player could always PICK one target to get their class-specialized info, in addition to picking WHICH secondary prediction to make, and WHICH target to use the secondary prediction on. In doing so, the player has exercised a choice, and has "personally contributed to the group information gathering", even if they were to copy/paste the results.

**Comment by doubledh** [comment]
I also propose that generally-static information, not be anyone's specialization. Then, someone won't feel bad that they "need" to reveal the element of an enemy that they obviously know the element of.

**Comment by ntseng** [comment]
### Accumulating
My biggest concern is that this converts the current communication mechanic (albeit poorly tuned) into a routing mechanic ("how do we get the important information the most quickly?") without increasing player agency in the replayability case. If the community decides `speed information is the most generally valuable` then one player volunteers to predict speed on their first turn of every fight and then every fight has a preamble of non-adaptive inputs. Even if we were able to design enemies in such a way that predicts are prioritized in different ways based on which enem(ies) are present (which I'm not confident of without large downstream balance effects), enemy team compositions are still static (and thus vulnerable to deciding beforehand what the predict route will be). From there the next logical step may be randomly generating enemy teams, but that doesn't escape reading from a tier list either and dilutes the opportunity to have targeted set pieces outside of boss battles.

### Unspecialized
I think having some situations where information is unreachable is actually good; otherwise the game becomes much more deterministic. Further, in my opinion, having information type more coupled with archetype leads to a slightly less arbitrary variety of self-induced challenge options ("can we beat the game without X character?" vs "can we beat the game without X predict?"). I think letting an archetype have a unique set of predicts achieves similar results (and probably mitigates the non-linear performance improvement of having more players a bit), but that commits us to making the tech to enforce a player not getting more than one type of prediction per turn (which may be worth it).

### Incremental
I have a lot of concerns about predicts having varying efficiencies. I do like that it allows for characterization of archetypes (although am still leaning toward "unique set" personally), however I'm not confident we can deliver good balance. In addition to increasing the number of entities that need to be balanced (from 6 to n sets of 6), I'm not sure each predict has a sensible intermediate state and would be wary to rely on AoE for that because the monolith boss battle (eg The Elkemist) is a situation we want to support. Some predicts (eg Move Names and Critical Hits) are fairly boolean in nature, so it wouldn't be important to have a specialist in those prediction types on a boss with a single enemy. However, the monolith boss is when we want to show off specialization the most, as to align mechanics with the narrative of difference in type triumphing over difference in scale.

**Reply by doubledh** [reply]
#Accumulating
Because different predict information is just innately/orthogonally different from each other (HP to crit is apples to oranges ... though we might be able to discuss if we CAN categorize the actionability or threat-gauging-ability of certain predicts), I don't believe the community will be ABLE to decide that any one predict is the meta-viable option. And even if they CAN, it will be a WHILE before they determine which one of the orthogonal predicts is the meta. 

And with information accumulation, even if there was a "speed-is-meta", you would have to pick another predict AFTER you exhausted all speed predicts. Then, there'd be another tossup between the orthogonal predicts. The community could create a tier list or flowchart of the most important information as an order by which to get data, but I feel like following that blindly would not always help players with the specific, random encounters that they are in. Also, tier lists and meta's themselves aren't the most evil that can come out of a game. Strategizing and coming up with dominant strategies is how people come up with guides, after all, and people still have fun playing games to guides. The forerunners of raids and guides also get a first-man-on-the-moon kick out of it, too.

#Unspecialized
I agree with limited-subset of predicts per archetype, it allows for adaptive choice, but the choice is still relevant to the core decision of archetype. I worry about "can we beat the game without X character" being discussed because I don't want to see a Thresh in every LoL draft, because people won't play a game without the guy D:< (well, not specifically Thresh, but you get the point that I don't think not-using-a-meta-character would be a well-meant challenge, especially self-induced)

#Efficiency
Because predicts are orthogonal, I actually don't think we actually need to balance them that well, at all. Players can build experience in figuring out which ones they find useful, or the ones they like. As for intermediate states, I agree that "partially revealing" information to resemble reduced efficiency is not very practical or applicable in a general case to each of our predicts. I think that "revealing information for 1 target for free per turn" is very applicable to all of our predicts, if we made each of our predicts only reveal info of a fixed number of targets in the normal case. Efficiency could be independent of Predict-Accumulation, where a player reveals X number of targets' info per turn and may have to recheck the information. However, because that would be a lot of REPETETIVE button pressing to recheck mostly-similar information, I would recommend having Predict-Accumulation so that way each turn a predict will add NEW information with nothing wasted on rechecking... and a universal view for predicts would be useful to display all that accumulated predict info.

**Reply by ntseng** [reply]
### Accumulating
Agreed that predicts aren't a boring optimization problem and that tier lists aren't the worst. My main claim is that I think there exists a state where distilling predicts is more fun than routing scouting.

### Unspecialized
I think we're a little less likely to see players bound to certain strategies for 2 reasons:
- The game isn't as directly competitive as League (for example)
- The game encourages stable groups, in which experimentation is usually more accepted

Known strats will likely have the same inertia, but I'm not as worried about players feeling restricted from experimenting.

### Efficiency
I think pairing accumulation with predict routing makes sense. Let's keep it in mind for if we do go the routing route.
